retrofit/src/main/java/retrofit/OkHttpCall.java

```diff
@@ -30,6 +30,7 @@
 
   private volatile com.squareup.okhttp.Call rawCall;
   private boolean executed; // Guarded by this.
```

Comment:
Which makes me think this could use an AtomicBooleanFieldUpdater for lock-free CAS changes